src/env.cr

```diff
@@ -32,10 +32,16 @@ module ENV
   # Overwrites existing environment variable if already present.
   # Returns *value* if successful, otherwise raises an exception.
   # If *value* is `nil`, the environment variable is deleted.
+  #
+  # If *key* or *value* contains a null-byte an `ArgumentError` is raised.
   def self.[]=(key : String, value : String?)
+    raise ArgumentError.new("Key contains null byte") if key.byte_index(0)
```

Comment:
You can simply do `key.check_no_null_byte`

Reply:
Same applies to `value`, of course

Reply:
Yes I had that at first, but then chose to tell if the key or the value is the culprit

Reply:
Ooooh... makes sense!